### ISSUE TITLE
[Package Issue]: AIMP.AIMP update is not "recorded" by winget

### ISSUE THREAD

**Issue by lackovic** [microsoft/winget-pkgs#83481]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Other

### Brief description of your issue

When running `winget upgrade -h --all`:

- `winget` reports the update is successful 
- I verify the version is indeed upgraded by opening the application

When running `winget upgrade -h --all` again `winget` incorrectly detects the old version as installed and install the new version again.

### Steps to reproduce

```
> winget install -e --id AIMP.AIMP
> winget upgrade -h --all
> winget upgrade -h --all
```

### Actual behavior

The application is upgraded twice.

### Expected behavior

The application is upgraded only once.

### Environment

```shell
Windows Package Manager v1.3.2691
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19043.2130
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.18.2691.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

### Screenshots and Logs

```
 > winget upgrade -h --all
Name                         Id            Version                 Available               Source
-------------------------------------------------------------------------------------------------
AIMP                         AIMP.AIMP     > 5.01.2358             5.03.2398               winget

(1/1) Found AIMP [AIMP.AIMP] Version 5.03.2398
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://aimp.ru/files/windows/builds/aimp_5.03.2398_w64.exe
  ██████████████████████████████  17.9 MB / 17.9 MB
Successfully verified installer hash
Starting package install...
Successfully installed
```

**Comment by Trenly** [comment]
What is shown when you run `winget list AIMP` ?

**Comment by lackovic** [comment]
> What is shown when you run `winget list AIMP` ?

```
Name Id        Version     Available Source
-------------------------------------------
AIMP AIMP.AIMP > 5.01.2358 5.03.2398 winget
```

**Comment by iDolmatov** [comment]
> aimp_5.03.2398_w64.exe

Most likely, a duplicate entry from the x86 version remained in the registry.
x64 HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\AIMP
x86 HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\AIMP
Or search by version number. Remove the duplicate AIMP section from the old version.

**Comment by Trenly** [comment]
Thank you for taking the time to report this issue. In the amount of time the issue has been open, there have been several updates to the WinGet CLI which may have helped mitigate some of these concerns. I encourage you to try out the [latest version](https://github.com/microsoft/winget-cli/releases/latest) of the CLI and see if your problem still persists. 

If additional problems persist, it would be extremely helpful if a new issue could be opened. Thanks!

###### Close with reason: Stale;
